prototyping_parallel/heat.py

```diff
@@ -257,18 +257,16 @@ def determine_gradient(V_g, u, flux):
         print('{rank} of {size}:running stupid loop'.format(rank=MPI.rank(MPI.comm_world), size=MPI.size(MPI.comm_world)))
         for vertex in fenics.vertices(mesh):  # TODO: this loop has no real purpose, however, if we do not execute it, the program hangs later on...
             flux(vertex.x(0), vertex.x(1))
```

Comment:
The code hangs if this loop is removed. The problem of hanged state when passing a FEniCS Function to an adapter function has been narrowed down to the case where a FEniCS Vector Function is passed to the adapter and then the evaluation of the function inside the adapter is attempted. The current commit hangs exactly at [this line](https://github.com/precice/fenics-adapter/pull/71/commits/47c79a9a17aa7c564af64cd4ae017751e59a3393#diff-c99807d6a2ee07096320e7eda2417448R106)